cpp/test/utils/metrics/log_metric_test.cc

```diff
@@ -0,0 +1,150 @@
+#include "src/utils/metrics/log_metric.hh"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <filesystem>
+#include <fstream>
+#include <regex>
+#include <string>
+#include <vector>
+
+#include "src/utils/logging.hh"
+
+namespace torchserve {
+class TSLogMetricTest : public ::testing::Test {
+ protected:
+  const std::string logfile_path{"test/resources/logging/test.log"};
```

Comment:
it is not a good way to write a log in resource. Backend log is stdio log.

Reply:
Will update the log path to `test/utils/metrics/metrics_test.log`

Reply:
Also, I considered the option of capturing the `stdout` for the unit test, but it appears that `gtest` does not offer any public API for capturing test `stdout` from what I could find. Although it does have an internal API for capturing `stdout` which may not be advisable to use in the test code:

1. https://github.com/google/googletest/blob/191ca1f3a9262b90a586ae2c2e8c742c3d867801/googletest/src/gtest-port.cc#L1117
2. https://stackoverflow.com/questions/3803465/how-to-capture-stdout-stderr-with-googletest

Therefore instead, I used log to file and then parse the file to validate the log output.